patroni/postgresql.py

```diff
@@ -933,6 +973,11 @@ def stop(self, mode='fast', block_callbacks=False, checkpoint=True, on_safepoint
 
     def _do_stop(self, mode, block_callbacks, checkpoint, on_safepoint):
         if not self.is_running():
+            pid = self.get_pid_with_lost_data_dir()
+            if pid > 0:
+                self.terminate_starting_postmaster(pid)
+                self.clean_postmaster_info()
```

Comment:
The on_safepoint callback is not called. We could pass a watchdog.disable here from ha.py:1018 and call on_safepoint, eliminating line 1020 and removing a TODO at line 1019.